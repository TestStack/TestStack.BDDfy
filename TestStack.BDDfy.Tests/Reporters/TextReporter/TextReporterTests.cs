using System;
using System.Runtime.CompilerServices;
using System.Text;
using ApprovalTests;
using NUnit.Framework;
using TestStack.BDDfy.Reporters;

namespace TestStack.BDDfy.Tests.Reporters.MarkDown
{
    [TestFixture]
    public class TextReporterTests
    {
        [Test]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ShouldProduceExpectedReport()
        {
            var stories = new ReportTestData().CreateMixContainingEachTypeOfOutcomeWithOneScenarioPerStory();
            var actual = new StringBuilder();

            foreach (var story in stories)
            {
                var textReporter = new TextReporter();
                textReporter.Process(story);
                actual.AppendLine(textReporter.ToString());
            }

            Approvals.Verify(actual.ToString(), StackTraceScrubber.Scrub);
        }

        [Test]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ShouldProduceExpectedTextWithExamples()
        {
            var stories = new ReportTestData().CreateTwoStoriesEachWithOneFailingScenarioAndOnePassingScenarioWithThreeStepsOfFiveMillisecondsAndEachHasTwoExamples();
            var actual = new StringBuilder();

            foreach (var story in stories)
            {
                var textReporter = new TextReporter();
                textReporter.Process(story);
                actual.AppendLine(textReporter.ToString());
            }

            Approvals.Verify(actual.ToString(), StackTraceScrubber.Scrub);
        }
    }
}
