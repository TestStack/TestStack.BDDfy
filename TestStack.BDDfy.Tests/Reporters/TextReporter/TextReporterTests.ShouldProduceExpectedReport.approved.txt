Story: Happiness
	As a person
	I want ice cream
	So that I can be happy

Scenario: Happy Path Scenario [for Happiness]
	Given a positive account balance
	When the account holder requests money
	Then money is dispensed


Story: Happiness
	As a person
	I want ice cream
	So that I can be happy

Scenario: Sad Path Scenario [for Happiness]
	Given a negative account balance             [Passed] 
	When the account holder requests money       [Passed] 
	Then no money is dispensed                   [Failed] [Boom] [Details at 1 below]

Exceptions:
  1. Boom
   at TestStack.BDDfy.Tests.Reporters.ReportTestData.GetFailingExecutionSteps() in ...\ReportTestData.cs


Story: Happiness
	As a person
	I want ice cream
	So that I can be happy

Scenario: Inconclusive Scenario [for Happiness]
	Given a negative account balance             [Passed] 
	When the account holder requests money       [Passed] 
	Then no money is dispensed                   [Inconclusive] 


Story: Happiness
	As a person
	I want ice cream
	So that I can be happy

Scenario: Not Implemented Scenario [for Happiness]
	Given a negative account balance             [Passed] 
	When the account holder requests money       [Passed] 
	Then no money is dispensed                   [Not implemented] 



Scenario: Happy Path Scenario [with no story]
	Given a positive account balance
	When the account holder requests money
	Then money is dispensed



Scenario: Sad Path Scenario [with no story]
	Given a negative account balance             [Passed] 
	When the account holder requests money       [Passed] 
	Then no money is dispensed                   [Failed] [Boom] [Details at 1 below]

Exceptions:
  1. Boom
   at TestStack.BDDfy.Tests.Reporters.ReportTestData.GetFailingExecutionSteps() in ...\ReportTestData.cs



Scenario: Inconclusive Scenario [with no story]
	Given a negative account balance             [Passed] 
	When the account holder requests money       [Passed] 
	Then no money is dispensed                   [Inconclusive] 



Scenario: Not Implemented Scenario [with no story]
	Given a negative account balance             [Passed] 
	When the account holder requests money       [Passed] 
	Then no money is dispensed                   [Not implemented] 


Story: Unhappy examples
	As an unhappy examples story
	I want to see failed steps
	So that I can diagnose what's wrong

Scenario: Example Scenario
	Given a <sign> account balance        
	When the account holder requests money
	Then money <action> dispensed         

Examples: 
| sign     | action | Result | Errors                                                                                                |
| positive | is     | Passed |                                                                                                       |
| negative | is not | Failed | Step: Then money <action> dispensed failed with exception: [Boom With New lines] [Details at 1 below] |

Exceptions:
  1. Boom With New lines
   at TestStack.BDDfy.Tests.Reporters.ReportTestData.GetScenarios(Boolean includeFailingScenario, Boolean includeExamples) in ...\ReportTestData.cs


Story: Happy Examples
	As a happy examples story
	I want a clean report with examples
	So that the report is clean and readable

Scenario: Example Scenario
	Given a <sign> account balance        
	When the account holder requests money
	Then money <action> dispensed         

Examples: 
| sign     | action |
| positive | is     |
| negative | is not |


